lib/rubygems/resolver/activation_request.rb

```diff
@@ -49,15 +49,28 @@ def development?
   # Downloads a gem at +path+ and returns the file path.
 
   def download path
-    if @spec.respond_to? :source
+    Gem.ensure_gem_subdirectories path
+
+    if @spec.respond_to? :sources
+      exception = nil
+      path = nil
+      @spec.sources.each{ |source|
+        begin
+          path = source.download full_spec, path
+        rescue exception
+          nil
+        end
+      }
```

Comment:
Wouldn't it be better to use `Array#find`? This avoids a nasty begin/rescue to a nil and losing the exception.